.github/workflows/test-lint.yml

```diff
@@ -0,0 +1,34 @@
+name: test-and-lint
+
+on: pull_request
+
+jobs:
+  test-and-lint:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.9.16
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.9.16'
```

Comment:
Should we use 3.11?

Reply:
A higher version makes sense, yes.

However, noticed that our Docker image has `python:3.9` as a base image as well and thus would like to address that in a separate PR.